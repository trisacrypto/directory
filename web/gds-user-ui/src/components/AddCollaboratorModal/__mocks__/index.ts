
export const collaboratorMockValue: any = {
    data: [
        {
            id: "1",
            name: "test",
            email: "test@local.dev",
            roles: ["admin"],
            created_at: "2021-01-01",
        },
    ],
};
