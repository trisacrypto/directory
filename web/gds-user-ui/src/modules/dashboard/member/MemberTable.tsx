import {
  Table,
  TableCaption,
  Tbody,
  Td,
  Th,
  Thead,
  Heading,
  Tr,
  Button,
  HStack,
  chakra,
  useColorModeValue,
} from '@chakra-ui/react';

import FormLayout from 'layouts/FormLayout';

import React from 'react';
import { Trans } from '@lingui/macro';
import { BsEye } from 'react-icons/bs';
import MemberSelectNetwork from './memberNetworkSelect';

const TableRow: React.FC = () => {
  return (
    <Tr>
      <Td>
        <chakra.span display="block"></chakra.span>
        <chakra.span display="block" fontSize="sm" color="gray.700"></chakra.span>
      </Td>
      <Td></Td>
      <Td></Td>
      <Td></Td>
      <Td></Td>
      <Td paddingY={0}>
        <HStack width="100%" justifyContent="center" alignItems="center">
          <Button
            color="blue"
            as={'a'}
            href={``}
            bg={'transparent'}
            _hover={{
              bg: 'transparent'
            }}
            _focus={{
              bg: 'transparent'
            }}>
            <BsEye fontSize="24px" />
          </Button>
        </HStack>
      </Td>
    </Tr>
  );
};

const TableRows: React.FC = () => {
  return (
    <>
      <TableRow />
    </>
  );
};

const MemberTable: React.FC = () => {
  const modalHandler = () => {
    console.log('modalHandler');
  };

  return (
    <FormLayout overflowX={'scroll'}>
      <Table variant="simple">
        <TableCaption placement="top" textAlign="end" p={0} m={0} mb={3} fontSize={20}>
          <HStack justify={'space-between'} mb={'10'}>
            <Heading size="md" color={'black'}>
              <Trans>Member List</Trans>
            </Heading>
            <Button
              minW="100px"
              onClick={modalHandler}
              bg={useColorModeValue('black', 'white')}
              _hover={{
                bg: useColorModeValue('black', 'white')
              }}
              color={useColorModeValue('white', 'black')}>
              <Trans>Export</Trans>
            </Button>
          </HStack>
          <MemberSelectNetwork />
        </TableCaption>
        <Thead>
          <Tr>
            <Th>
              <Trans>Member Name</Trans>
            </Th>
            <Th>
              <Trans>Joined</Trans>
            </Th>
            <Th>
              <Trans>Last Updated</Trans>
            </Th>
            <Th>
              <Trans>Network</Trans>
            </Th>
            <Th>
              <Trans>Status</Trans>
            </Th>
            <Th textAlign="center">
              <Trans>Actions</Trans>
            </Th>
          </Tr>
        </Thead>
        <Tbody>
          <TableRows />
        </Tbody>
      </Table>
    </FormLayout>
  );
};
export default MemberTable;
