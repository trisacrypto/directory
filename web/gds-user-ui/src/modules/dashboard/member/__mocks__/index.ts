export const mainnetMembersMockValue: any = {
  vasps: [
    {
      id: 'fe29b582-20b3-4b75-acb2-4d014af25f28',
      registered_directory: 'vaspdirectory.net',
      common_name: 'mainnet.firecoinex.co',
      endpoint: 'mainnet.firecoinex.co:443',
      name: 'FireCoin Exchange',
      website: 'https://www.firecoinex.co/',
      country: 'SG',
      business_category: 'BUSINESS_ENTITY',
      vasp_categories: ['OTC'],
      verified_on: '2022-04-20T10:45:33Z',
      status: 'VERIFIED',
      first_listed: '2022-04-20T11:10:06Z',
      last_updated: '2022-11-19T09:32:45Z'
    },
    {
      id: '52842ac3-7d9d-4520-8bfb-e6083dbdc8aa',
      registered_directory: 'vaspdirectory.net',
      common_name: 'mainnet.testmachine.com',
      endpoint: 'mainnet.testmachine.com:443',
      name: 'Test Machine',
      website: 'https://testmachine.com',
      country: 'US',
      business_category: 'BUSINESS_ENTITY',
      vasp_categories: ['Exchange', 'Other'],
      verified_on: '2023-03-22T03:55:00Z',
      status: 'VERIFIED',
      first_listed: '2023-03-22T04:22:18Z',
      last_updated: '2023-05-10T04:44:00Z'
    },
    {
      id: 'ec6f2056-726b-4e8c-a916-f7359f6f5581',
      registered_directory: 'vaspdirectory.net',
      common_name: 'mainnet.newcoinex.ai',
      endpoint: 'mainnet.newcoinex.ai:8221',
      name: 'New Coin Exchange',
      website: 'https://newcoinex.ai/',
      country: 'US',
      business_category: 'BUSINESS_ENTITY',
      vasp_categories: ['Other'],
      verified_on: '2023-01-20T12:55:03Z',
      status: 'VERIFIED',
      first_listed: '2023-01-20T14:10:26Z',
      last_updated: '2023-06-18T10:30:01Z'
    }
  ],
  next_page_token: ''
};

export const testnetMembersMockValue: any = {
  vasps: [
    {
      id: '1b99f17d-3441-4885-b9df-8f3475f7e1b4',
      registered_directory: 'trisatest.net',
      common_name: 'trisa-travelrule.sendcoin.io',
      endpoint: 'trisa-travelrule.sendcoin.io:443',
      name: 'SendCoin VASP',
      website: 'https://www.sendcoin.io/',
      country: 'SG',
      business_category: 'BUSINESS_ENTITY',
      vasp_categories: ['OTC'],
      verified_on: '2022-02-10T17:12:23Z',
      status: 'VERIFIED',
      first_listed: '2022-02-10T13:43:55Z',
      last_updated: '2023-04-09T04:02:50Z'
    },
    {
      id: 'aa4c6714-49dc-411c-adfa-b8edc4d58cd7',
      registered_directory: 'trisatest.net',
      common_name: 'testing.example.com',
      endpoint: 'testing.example.com:443',
      name: 'Example Crypto',
      website: 'https://example.com',
      country: 'US',
      business_category: 'BUSINESS_ENTITY',
      vasp_categories: ['Exchange', 'Other'],
      verified_on: '2021-12-07T20:22:00Z',
      status: 'VERIFIED',
      first_listed: '2021-12-01T23:22:18Z',
      last_updated: '2023-01-23T19:19:43Z'
    },
    {
      id: '688059d6-9d14-4b49-8435-f641ba1dec3a',
      registered_directory: 'trisatest.net',
      common_name: 'testnet.spudcoin.ai',
      endpoint: 'testnet.spudcoin.ai:8221',
      name: 'SpudCoin',
      website: 'https://spudcoin.ai/',
      country: 'US',
      business_category: 'BUSINESS_ENTITY',
      vasp_categories: ['Other'],
      verified_on: '2021-07-29T19:11:03Z',
      status: 'VERIFIED',
      first_listed: '2021-07-23T17:10:26Z',
      last_updated: '2022-12-27T18:59:01Z'
    },
    {
      id: '62291255-1ea2-4932-8248-22af4abde298',
      registered_directory: 'trisatest.net',
      common_name: 'test.signal.co.fr',
      endpoint: 'test.signal.co.fr:9212',
      name: 'Signal Coin France',
      website: 'https://ciphertrace.com/',
      country: 'US',
      business_category: 'BUSINESS_ENTITY',
      vasp_categories: ['Other'],
      verified_on: '2021-06-23T17:46:10Z',
      status: 'VERIFIED',
      first_listed: '2021-06-23T17:12:00Z',
      last_updated: '2022-12-27T18:59:22Z'
    },
    {
      id: 'dae1555d-e4cf-4bfb-9858-9b86db71ccb6',
      registered_directory: 'trisatest.net',
      common_name: 'testnet.bitfriend.link',
      endpoint: 'testnet.bitfriend.link:443',
      name: 'BitFriendly',
      website: 'https://bitfriend.link/',
      country: 'US',
      business_category: 'BUSINESS_ENTITY',
      vasp_categories: ['Kiosk'],
      verified_on: '2021-09-14T10:40:30Z',
      status: 'VERIFIED',
      first_listed: '2021-09-01T19:46:04Z',
      last_updated: '2023-01-23T19:29:43Z'
    }
  ],
  next_page_token: 'mLB9CU8O8xQj2XEyjAtlfvTj9imoXnLv/1p8fTLchTg='
};

export const getMockValue = (network: string): any => {
  switch (network) {
    case 'mainnet':
      return mainnetMembersMockValue;
    case 'testnet':
      return testnetMembersMockValue;
    default:
      return mainnetMembersMockValue;
  }
};
