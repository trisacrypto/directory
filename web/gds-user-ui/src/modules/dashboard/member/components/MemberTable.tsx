import { Table, TableCaption, Tbody, Th, Thead, Tr } from '@chakra-ui/react';

import React from 'react';
import { Trans } from '@lingui/macro';

import MemberTableList from './MemberTableList';

// interface MemberTableProps {
//   data: any;
// }

const MemberTable: React.FC = () => {
  return (
    <Table variant="simple">
      <TableCaption placement="top" textAlign="end" p={0} m={0} mb={3} fontSize={20}></TableCaption>
      <Thead>
        <Tr>
          <Th data-testid="name-header">
            <Trans>Member Name</Trans>
          </Th>
          <Th data-testid="joined-header">
            <Trans>Joined</Trans>
          </Th>
          <Th data-testid="last-updated-header">
            <Trans>Last Updated</Trans>
          </Th>
          <Th data-testid="network-header">
            <Trans>Network</Trans>
          </Th>
          <Th data-testid="status-header">
            <Trans>Status</Trans>
          </Th>
          <Th textAlign="center" data-testid="actions-header">
            <Trans>Actions</Trans>
          </Th>
        </Tr>
      </Thead>
      <Tbody>
        <MemberTableList />
      </Tbody>
    </Table>
  );
};
export default MemberTable;
