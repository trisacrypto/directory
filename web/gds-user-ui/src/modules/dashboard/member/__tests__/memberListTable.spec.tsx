import { act, render, screen } from 'utils/test-utils';
import MemberTable from "../MemberTable";
import { dynamicActivate } from 'utils/i18nLoaderHelper';
import { mainnetMembersMockValue } from '../__mocks__';

function renderComponent() {
   return render(<MemberTable data={mainnetMembersMockValue} />);
}

describe('MemberTable', () => {
    beforeAll(() => {
        act(() => {
          dynamicActivate('en');
        });
      });

      it('should render', () => {
        const { container } = renderComponent();
        expect(container).toMatchSnapshot();
      })

        it('should render the correct table headers', () => {
            renderComponent();
            const header = screen.getByText(/Member Name/i);
            const joined = screen.getByText(/Joined/i);
            const lastUpdate = screen.getByText(/Last Updated/i);
            // const network = screen.getByText(/Network/i);
            const status = screen.getByText(/Status/i);
            expect(header).toBeInTheDocument();
            expect(joined).toBeInTheDocument();
            expect(lastUpdate).toBeInTheDocument();
            // expect(network).toBeInTheDocument();
            expect(status).toBeInTheDocument();
        });
});
