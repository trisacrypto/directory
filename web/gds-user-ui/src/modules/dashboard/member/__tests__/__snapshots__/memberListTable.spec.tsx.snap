// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MemberTable should render 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background: var(--chakra-colors-white);
  border: 2px solid #E5EDF1;
  border-radius: 10px;
  padding: var(--chakra-space-3);
  overflow-x: scroll;
}

.emotion-0>*:not(style)~*:not(style) {
  margin-top: var(--chakra-space-3-5);
  -webkit-margin-end: 0px;
  margin-inline-end: 0px;
  margin-bottom: 0px;
  -webkit-margin-start: 0px;
  margin-inline-start: 0px;
}

@media screen and (min-width: 48em) {
  .emotion-0 {
    padding: var(--chakra-space-9);
  }
}

.emotion-1 {
  font-variant-numeric: lining-nums tabular-nums;
  border-collapse: collapse;
  width: var(--chakra-sizes-full);
}

.emotion-2 {
  margin-top: var(--chakra-space-4);
  font-family: var(--chakra-fonts-heading);
  text-align: end;
  font-weight: var(--chakra-fontWeights-medium);
  -webkit-padding-start: var(--chakra-space-6);
  padding-inline-start: var(--chakra-space-6);
  -webkit-padding-end: var(--chakra-space-6);
  padding-inline-end: var(--chakra-space-6);
  padding-top: var(--chakra-space-2);
  padding-bottom: var(--chakra-space-2);
  font-size: 20px;
  color: var(--chakra-colors-gray-600);
  caption-side: top;
  padding: 0px;
  margin: 0px;
  margin-bottom: var(--chakra-space-3);
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin-bottom: var(--chakra-space-10);
}

.emotion-3>*:not(style)~*:not(style) {
  margin-top: 0px;
  -webkit-margin-end: 0px;
  margin-inline-end: 0px;
  margin-bottom: 0px;
  -webkit-margin-start: 0.5rem;
  margin-inline-start: 0.5rem;
}

.emotion-4 {
  font-family: var(--chakra-fonts-heading);
  font-weight: var(--chakra-fontWeights-bold);
  font-size: var(--chakra-fontSizes-xl);
  line-height: 1.2;
  color: var(--chakra-colors-black);
}

.emotion-5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: relative;
  white-space: nowrap;
  vertical-align: middle;
  outline: 2px solid transparent;
  outline-offset: 2px;
  width: auto;
  line-height: 1.2;
  border-radius: var(--chakra-radii-md);
  font-weight: var(--chakra-fontWeights-semibold);
  transition-property: var(--chakra-transition-property-common);
  transition-duration: var(--chakra-transition-duration-normal);
  height: var(--chakra-sizes-10);
  min-width: 100px;
  font-size: var(--chakra-fontSizes-md);
  -webkit-padding-start: var(--chakra-space-4);
  padding-inline-start: var(--chakra-space-4);
  -webkit-padding-end: var(--chakra-space-4);
  padding-inline-end: var(--chakra-space-4);
  background: var(--chakra-colors-black);
  color: var(--chakra-colors-white);
}

.emotion-5:focus,
.emotion-5[data-focus] {
  box-shadow: var(--chakra-shadows-outline);
}

.emotion-5[disabled],
.emotion-5[aria-disabled=true],
.emotion-5[data-disabled] {
  opacity: 0.4;
  cursor: not-allowed;
  box-shadow: var(--chakra-shadows-none);
}

.emotion-5:hover,
.emotion-5[data-hover] {
  background: var(--chakra-colors-black);
}

.emotion-5:active,
.emotion-5[data-active] {
  background: var(--chakra-colors-gray-300);
}

.emotion-6 {
  width: 100%;
  position: relative;
}

.emotion-7 {
  display: block;
  text-align: start;
  font-size: var(--chakra-fontSizes-md);
  -webkit-margin-end: var(--chakra-space-3);
  margin-inline-end: var(--chakra-space-3);
  margin-bottom: var(--chakra-space-2);
  font-weight: var(--chakra-fontWeights-medium);
  transition-property: var(--chakra-transition-property-common);
  transition-duration: var(--chakra-transition-duration-normal);
  opacity: 1;
}

.emotion-7[disabled],
.emotion-7[aria-disabled=true],
.emotion-7[data-disabled] {
  opacity: 0.4;
}

.emotion-8 {
  width: 100%;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  position: relative;
}

.emotion-9 {
  -webkit-padding-end: 2rem;
  padding-inline-end: 2rem;
  width: 100%;
  min-width: 0px;
  outline: 2px solid transparent;
  outline-offset: 2px;
  position: relative;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  transition-property: var(--chakra-transition-property-common);
  transition-duration: var(--chakra-transition-duration-normal);
  background: inherit;
  padding-bottom: 1px;
  line-height: var(--chakra-lineHeights-normal);
  font-size: var(--chakra-fontSizes-md);
  -webkit-padding-start: var(--chakra-space-4);
  padding-inline-start: var(--chakra-space-4);
  height: var(--chakra-sizes-10);
  border-radius: var(--chakra-radii-md);
  border: 1px solid;
  border-color: inherit;
}

.emotion-9>option,
.emotion-9>optgroup {
  background: var(--chakra-colors-white);
}

.emotion-9:hover,
.emotion-9[data-hover] {
  border-color: var(--chakra-colors-gray-300);
}

.emotion-9[aria-readonly=true],
.emotion-9[readonly],
.emotion-9[data-readonly] {
  box-shadow: none!important;
  -webkit-user-select: all;
  -moz-user-select: all;
  -ms-user-select: all;
  user-select: all;
}

.emotion-9[disabled],
.emotion-9[aria-disabled=true],
.emotion-9[data-disabled] {
  opacity: 0.4;
  cursor: not-allowed;
}

.emotion-9[aria-invalid=true],
.emotion-9[data-invalid] {
  border-color: #E53E3E;
  box-shadow: 0 0 0 1px #E53E3E;
}

.emotion-9:focus,
.emotion-9[data-focus] {
  z-index: unset;
  border-color: #3182ce;
  box-shadow: 0 0 0 1px #3182ce;
}

.emotion-10 {
  width: 1.5rem;
  height: 100%;
  right: 0.5rem;
  position: absolute;
  color: currentColor;
  font-size: 1.25rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  pointer-events: none;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -moz-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.emotion-10[disabled],
.emotion-10[aria-disabled=true],
.emotion-10[data-disabled] {
  opacity: 0.5;
}

.emotion-13 {
  font-family: var(--chakra-fonts-heading);
  font-weight: var(--chakra-fontWeights-bold);
  text-transform: uppercase;
  letter-spacing: var(--chakra-letterSpacings-wider);
  text-align: start;
  -webkit-padding-start: var(--chakra-space-6);
  padding-inline-start: var(--chakra-space-6);
  -webkit-padding-end: var(--chakra-space-6);
  padding-inline-end: var(--chakra-space-6);
  padding-top: var(--chakra-space-3);
  padding-bottom: var(--chakra-space-3);
  line-height: var(--chakra-lineHeights-4);
  font-size: var(--chakra-fontSizes-xs);
  color: var(--chakra-colors-gray-600);
  border-bottom: var(--chakra-borders-1px);
  border-color: var(--chakra-colors-gray-100);
}

.emotion-13[data-is-numeric=true] {
  text-align: end;
}

.emotion-18 {
  font-family: var(--chakra-fonts-heading);
  font-weight: var(--chakra-fontWeights-bold);
  text-transform: uppercase;
  letter-spacing: var(--chakra-letterSpacings-wider);
  text-align: center;
  -webkit-padding-start: var(--chakra-space-6);
  padding-inline-start: var(--chakra-space-6);
  -webkit-padding-end: var(--chakra-space-6);
  padding-inline-end: var(--chakra-space-6);
  padding-top: var(--chakra-space-3);
  padding-bottom: var(--chakra-space-3);
  line-height: var(--chakra-lineHeights-4);
  font-size: var(--chakra-fontSizes-xs);
  color: var(--chakra-colors-gray-600);
  border-bottom: var(--chakra-borders-1px);
  border-color: var(--chakra-colors-gray-100);
}

.emotion-18[data-is-numeric=true] {
  text-align: end;
}

.emotion-21 {
  text-align: start;
  -webkit-padding-start: var(--chakra-space-6);
  padding-inline-start: var(--chakra-space-6);
  -webkit-padding-end: var(--chakra-space-6);
  padding-inline-end: var(--chakra-space-6);
  padding-top: var(--chakra-space-4);
  padding-bottom: var(--chakra-space-4);
  line-height: var(--chakra-lineHeights-5);
  border-bottom: var(--chakra-borders-1px);
  border-color: var(--chakra-colors-gray-100);
}

.emotion-21[data-is-numeric=true] {
  text-align: end;
}

.emotion-22 {
  display: block;
}

.emotion-27 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  vertical-align: top;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  max-width: 100%;
  font-weight: var(--chakra-fontWeights-medium);
  line-height: 1.2;
  outline: 2px solid transparent;
  outline-offset: 2px;
  min-height: 1.5rem;
  min-width: 1.5rem;
  font-size: var(--chakra-fontSizes-sm);
  border-radius: var(--chakra-radii-md);
  -webkit-padding-start: var(--chakra-space-2);
  padding-inline-start: var(--chakra-space-2);
  -webkit-padding-end: var(--chakra-space-2);
  padding-inline-end: var(--chakra-space-2);
  background: var(--chakra-colors-green-400);
  color: var(--chakra-colors-white);
}

.emotion-27:focus,
.emotion-27[data-focus] {
  box-shadow: var(--chakra-shadows-outline);
}

.emotion-28 {
  text-align: start;
  -webkit-padding-start: var(--chakra-space-6);
  padding-inline-start: var(--chakra-space-6);
  -webkit-padding-end: var(--chakra-space-6);
  padding-inline-end: var(--chakra-space-6);
  padding-top: 0px;
  padding-bottom: 0px;
  line-height: var(--chakra-lineHeights-5);
  border-bottom: var(--chakra-borders-1px);
  border-color: var(--chakra-colors-gray-100);
}

.emotion-28[data-is-numeric=true] {
  text-align: end;
}

.emotion-29 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
}

.emotion-29>*:not(style)~*:not(style) {
  margin-top: 0px;
  -webkit-margin-end: 0px;
  margin-inline-end: 0px;
  margin-bottom: 0px;
  -webkit-margin-start: 0.5rem;
  margin-inline-start: 0.5rem;
}

.emotion-30 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: relative;
  white-space: nowrap;
  vertical-align: middle;
  outline: 2px solid transparent;
  outline-offset: 2px;
  width: auto;
  line-height: 1.2;
  border-radius: var(--chakra-radii-md);
  font-weight: var(--chakra-fontWeights-semibold);
  transition-property: var(--chakra-transition-property-common);
  transition-duration: var(--chakra-transition-duration-normal);
  height: var(--chakra-sizes-10);
  min-width: var(--chakra-sizes-10);
  font-size: var(--chakra-fontSizes-md);
  -webkit-padding-start: var(--chakra-space-4);
  padding-inline-start: var(--chakra-space-4);
  -webkit-padding-end: var(--chakra-space-4);
  padding-inline-end: var(--chakra-space-4);
  background: var(--chakra-colors-transparent);
  color: blue;
}

.emotion-30:focus,
.emotion-30[data-focus] {
  background: var(--chakra-colors-transparent);
}

.emotion-30[disabled],
.emotion-30[aria-disabled=true],
.emotion-30[data-disabled] {
  opacity: 0.4;
  cursor: not-allowed;
  box-shadow: var(--chakra-shadows-none);
}

.emotion-30:hover,
.emotion-30[data-hover] {
  background: var(--chakra-colors-transparent);
}

.emotion-30:active,
.emotion-30[data-active] {
  background: var(--chakra-colors-gray-300);
}

<div>
  <div
    class="chakra-stack emotion-0"
  >
    <table
      class="chakra-table emotion-1"
      role="table"
    >
      <caption
        class="emotion-2"
      >
        <div
          class="chakra-stack emotion-3"
        >
          <h2
            class="chakra-heading emotion-4"
          >
            Member List
          </h2>
          <button
            class="chakra-button emotion-5"
            type="button"
          >
            Export
          </button>
        </div>
        <div
          class="chakra-form-control emotion-6"
          role="group"
        >
          <label
            class="chakra-form__label emotion-7"
            for="field-1"
            id="field-1-label"
          >
            Select Network
          </label>
          <div
            class="chakra-select__wrapper emotion-8"
          >
            <select
              class="chakra-select emotion-9"
              id="field-1"
            >
              <option
                value="mainnet"
              >
                MainNet
              </option>
              <option
                value="testnet"
              >
                TestNet
              </option>
            </select>
            <div
              class="chakra-select__icon-wrapper emotion-10"
            >
              <svg
                aria-hidden="true"
                class="chakra-select__icon"
                focusable="false"
                role="presentation"
                style="width: 1em; height: 1em; color: currentColor;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </div>
        </div>
      </caption>
      <thead
        class="emotion-11"
      >
        <tr
          class="emotion-11"
          role="row"
        >
          <th
            class="emotion-13"
            data-testid="name-header"
          >
            Member Name
          </th>
          <th
            class="emotion-13"
            data-testid="joined-header"
          >
            Joined
          </th>
          <th
            class="emotion-13"
            data-testid="last-updated-header"
          >
            Last Updated
          </th>
          <th
            class="emotion-13"
            data-testid="network-header"
          >
            Network
          </th>
          <th
            class="emotion-13"
            data-testid="status-header"
          >
            Status
          </th>
          <th
            class="emotion-18"
            data-testid="actions-header"
          >
            Actions
          </th>
        </tr>
      </thead>
      <tbody
        class="emotion-11"
      >
        <tr
          class="emotion-11"
          role="row"
        >
          <td
            class="emotion-21"
            role="gridcell"
          >
            <span
              class="emotion-22"
            >
              FireCoin Exchange
            </span>
          </td>
          <td
            class="emotion-21"
            role="gridcell"
          >
            Apr 20, 2022
          </td>
          <td
            class="emotion-21"
            role="gridcell"
          >
            Nov 19, 2022
          </td>
          <td
            class="emotion-21"
            role="gridcell"
          >
            MainNet
          </td>
          <td
            class="emotion-21"
            role="gridcell"
          >
            <span
              class="emotion-27"
            >
              VERIFIED
            </span>
          </td>
          <td
            class="emotion-28"
            role="gridcell"
          >
            <div
              class="chakra-stack emotion-29"
            >
              <a
                class="chakra-button emotion-30"
                href=""
              >
                <svg
                  fill="currentColor"
                  font-size="24px"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 16 16"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M16 8s-3-5.5-8-5.5S0 8 0 8s3 5.5 8 5.5S16 8 16 8zM1.173 8a13.133 13.133 0 0 1 1.66-2.043C4.12 4.668 5.88 3.5 8 3.5c2.12 0 3.879 1.168 5.168 2.457A13.133 13.133 0 0 1 14.828 8c-.058.087-.122.183-.195.288-.335.48-.83 1.12-1.465 1.755C11.879 11.332 10.119 12.5 8 12.5c-2.12 0-3.879-1.168-5.168-2.457A13.134 13.134 0 0 1 1.172 8z"
                  />
                  <path
                    d="M8 5.5a2.5 2.5 0 1 0 0 5 2.5 2.5 0 0 0 0-5zM4.5 8a3.5 3.5 0 1 1 7 0 3.5 3.5 0 0 1-7 0z"
                  />
                </svg>
              </a>
            </div>
          </td>
        </tr>
        <tr
          class="emotion-11"
          role="row"
        >
          <td
            class="emotion-21"
            role="gridcell"
          >
            <span
              class="emotion-22"
            >
              Test Machine
            </span>
          </td>
          <td
            class="emotion-21"
            role="gridcell"
          >
            Mar 22, 2023
          </td>
          <td
            class="emotion-21"
            role="gridcell"
          >
            May 10, 2023
          </td>
          <td
            class="emotion-21"
            role="gridcell"
          >
            MainNet
          </td>
          <td
            class="emotion-21"
            role="gridcell"
          >
            <span
              class="emotion-27"
            >
              VERIFIED
            </span>
          </td>
          <td
            class="emotion-28"
            role="gridcell"
          >
            <div
              class="chakra-stack emotion-29"
            >
              <a
                class="chakra-button emotion-30"
                href=""
              >
                <svg
                  fill="currentColor"
                  font-size="24px"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 16 16"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M16 8s-3-5.5-8-5.5S0 8 0 8s3 5.5 8 5.5S16 8 16 8zM1.173 8a13.133 13.133 0 0 1 1.66-2.043C4.12 4.668 5.88 3.5 8 3.5c2.12 0 3.879 1.168 5.168 2.457A13.133 13.133 0 0 1 14.828 8c-.058.087-.122.183-.195.288-.335.48-.83 1.12-1.465 1.755C11.879 11.332 10.119 12.5 8 12.5c-2.12 0-3.879-1.168-5.168-2.457A13.134 13.134 0 0 1 1.172 8z"
                  />
                  <path
                    d="M8 5.5a2.5 2.5 0 1 0 0 5 2.5 2.5 0 0 0 0-5zM4.5 8a3.5 3.5 0 1 1 7 0 3.5 3.5 0 0 1-7 0z"
                  />
                </svg>
              </a>
            </div>
          </td>
        </tr>
        <tr
          class="emotion-11"
          role="row"
        >
          <td
            class="emotion-21"
            role="gridcell"
          >
            <span
              class="emotion-22"
            >
              New Coin Exchange
            </span>
          </td>
          <td
            class="emotion-21"
            role="gridcell"
          >
            Jan 20, 2023
          </td>
          <td
            class="emotion-21"
            role="gridcell"
          >
            Jun 18, 2023
          </td>
          <td
            class="emotion-21"
            role="gridcell"
          >
            MainNet
          </td>
          <td
            class="emotion-21"
            role="gridcell"
          >
            <span
              class="emotion-27"
            >
              VERIFIED
            </span>
          </td>
          <td
            class="emotion-28"
            role="gridcell"
          >
            <div
              class="chakra-stack emotion-29"
            >
              <a
                class="chakra-button emotion-30"
                href=""
              >
                <svg
                  fill="currentColor"
                  font-size="24px"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 16 16"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M16 8s-3-5.5-8-5.5S0 8 0 8s3 5.5 8 5.5S16 8 16 8zM1.173 8a13.133 13.133 0 0 1 1.66-2.043C4.12 4.668 5.88 3.5 8 3.5c2.12 0 3.879 1.168 5.168 2.457A13.133 13.133 0 0 1 14.828 8c-.058.087-.122.183-.195.288-.335.48-.83 1.12-1.465 1.755C11.879 11.332 10.119 12.5 8 12.5c-2.12 0-3.879-1.168-5.168-2.457A13.134 13.134 0 0 1 1.172 8z"
                  />
                  <path
                    d="M8 5.5a2.5 2.5 0 1 0 0 5 2.5 2.5 0 0 0 0-5zM4.5 8a3.5 3.5 0 1 1 7 0 3.5 3.5 0 0 1-7 0z"
                  />
                </svg>
              </a>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
